stagecraft/apps/collectors/management/commands/import_collector_config.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import print_function
+
+import json
+import functools
+import hashlib
+import os
+
+from collections import defaultdict
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+
+from stagecraft.apps.collectors.models import (
+    Collector, CollectorType, DataSource, Provider
+)
+from stagecraft.apps.datasets.models import DataSet
+
+
+entrypoint_information = {
+    'performanceplatform.collector.ga': {
+        'credentials': 'ga-performanceplatform',
+        'repeat': 'daily',
+    },
+    'performanceplatform.collector.webtrends.reports': {
+        'credentials': {
+            'nas-applications': 'webtrends-national-apprenticeship-scheme',
+            'nhs-choices': 'webtrends-nhs-choices',
+        },
+        'repeat': 'daily',
+    },
+    'performanceplatform.collector.webtrends.keymetrics': {
+        'credentials': {
+            'nas-applications': 'webtrends-national-apprenticeship-scheme',
+            'nhs-choices': 'webtrends-nhs-choices',
+        },
+        'repeat': 'hourly',
+    },
+    'performanceplatform.collector.ga.trending': {
+        'credentials': 'ga-performanceplatform',
+        'repeat': 'daily',
+    },
+    'performanceplatform.collector.ga.contrib.content.table': {
+        'credentials': 'ga-performanceplatform',
+        'repeat': 'daily',
+    },
+    'performanceplatform.collector.ga.realtime': {
+        'credentials': 'ga-performanceplatform',
+        'repeat': '2minute',
+    },
+    'performanceplatform.collector.pingdom': {
+        'credentials': 'pingdom-performanceplatform',
+        'repeat': 'hourly',
+    },
+    'performanceplatform.collector.piwik.core': {
+        'credentials': 'piwik-fco',
+        'repeat': 'daily',
+    },
+    'performanceplatform.collector.piwik.realtime': {
+        'credentials': 'piwik-fco',
+        'repeat': '2minute',
+    },
+    'performanceplatform.collector.gcloud': {
+        'credentials': 'gcloud',
+        'repeat': 'hourly',
+    },
+}
+
+
+class Command(BaseCommand):
+    args = '<config_dir>'
+    help = 'Import collector configuration from repo.'
+
+    def handle(self, *args, **options):
+        if not len(args):
+            raise CommandError('No config directory given')
+
+        queries_dir = os.path.join(args[0], 'queries')
+        config = self.load_config(queries_dir)
+        by_slug = self.sort_by_slug(config)
+
+        self.import_config(by_slug)
+
+    def import_config(self, by_slug):
+        Collector.objects.filter(slug__iregex=r'^.+-[0-9a-f]{8}$').delete()
```

Comment:
Is this just deleting existing collectors so the script doesnt keep adding them when run multiple times? Its confusing what it is doing